openassessment/xblock/staff_area_mixin.py

```diff
@@ -226,7 +235,77 @@ def render_student_info(self, data, suffix=''):  # pylint: disable=W0613
             return self.render_assessment(path, context)
 
         except PeerAssessmentInternalError:
-            return self.render_error(self._(u"Error finding assessment workflow cancellation."))
+            return self.render_error(self._(u"Error getting learner information."))
+
+    @XBlock.handler
+    @require_course_staff("STUDENT_GRADE")
+    def render_staff_grade_form(self, data, suffix=''):  # pylint: disable=W0613
+        """
+        Renders a form to staff-grade the next available learner submission.
+
+        Must be course staff to render this view.
+        """
+        try:
+            student_item_dict = self.get_student_item_dict()
+            course_id = student_item_dict.get('course_id')
+            item_id = student_item_dict.get('item_id')
+            staff_id = student_item_dict['student_id']
+
+            # Note that this will check out a submission for grading by the specified staff member.
+            # If no submissions are available for graidng, will reutrn None.
+            submission_to_assess = staff_api.get_submission_to_assess(course_id, item_id, staff_id)
+
+            if submission_to_assess is not None:
+                submission = submission_api.get_submission_and_student(submission_to_assess['uuid'])
+                if submission:
+                    anonymous_student_id = submission['student_item']['student_id']
+                    submission_context = self.get_student_submission_context(
+                        self.get_username(anonymous_student_id), submission
+                    )
+                    path = 'openassessmentblock/staff_area/oa_staff_grade_learners_assessment.html'
+                    return self.render_assessment(path, submission_context)
+                else:
+                    return self.render_error(self._(u"Error loading the checked out learner response."))
+            else:
+                return self.render_error(self._(u"No more learner responses can be graded at this time."))
```

Comment:
What causes this error? Would it be more precise/informative to rewrite this error message as "No other learner responses are available for grading at this time."?